content/en/blog/_posts/2024-08-14-Kubernetes-v1-31-Release.md

```diff
@@ -0,0 +1,289 @@
+
+This work was done as part of [KEP #4680](https://github.com/kubernetes/enhancements/issues/4680) by [SIG Node](https://github.com/kubernetes/community/tree/master/sig-node).
+
+
+### Authorize with field and label selectors going Alpha in 1.31
+
+This feature allows webhook authorizers and future (but not currently designed) in-tree authorizers to allow list and watch requests based on field and label selectors.  It is now possible for an authorizer to express: this user cannot list all pods, but can list all pods where `.spec.nodeName=foo`.  Combined with CRD field selectors (moving to beta in 1.31), it is possible to write more secure per-node extensions.
+    
+This work was done as part of [KEP #4601](https://github.com/kubernetes/enhancements/issues/4601) by [SIG Auth](https://github.com/kubernetes/community/tree/master/sig-auth).
+    
+
+### Only allow anonymous auth for configured endpoints going Alpha in 1.31
+
+By enabling the feature gate `AnonymousAuthConfigurableEndpoints` users can now use the authentication configuration file to configure the endpoints that can be accessed by anonymous requests. This allows users to protect themselves against RBAC misconfigurations that can give anonymous users broad access to the cluster.
+    
+
+This work was done as a part of [KEP #4633](https://github.com/kubernetes/enhancements/issues/4633) and lead by [SIG Auth](https://github.com/kubernetes/community/tree/master/sig-auth).
+    
+## Graduations, deprecations, and removals in 1.31
+    
+
+### Graduations to Stable
+
+This lists all the features that graduated to stable (also known as _general availability_). For a full list of updates including new features and graduations from alpha to beta, see the release notes.
+
+This release includes a total of 11 enhancements promoted to Stable:
+
+* [PersistentVolume last phase transition time](	https://github.com/kubernetes/enhancements/issues/3762)
+* [Metric cardinality enforcement](	https://github.com/kubernetes/enhancements/issues/2305)
+* [Kube-proxy improved ingress connectivity reliability](	https://github.com/kubernetes/enhancements/issues/3836)
+* [Add CDI devices to device plugin API](	https://github.com/kubernetes/enhancements/issues/4009)
+* [Move cgroup v1 support into maintenance mode](	https://github.com/kubernetes/enhancements/issues/4569)
+* [AppArmor support](	https://github.com/kubernetes/enhancements/issues/24)
+* [PodHealthyPolicy for PodDisruptionBudget](	https://github.com/kubernetes/enhancements/issues/3017)
+* [Retriable and non-retriable Pod failures for Jobs](	https://github.com/kubernetes/enhancements/issues/3329)
+* [Elastic Indexed Jobs](	https://github.com/kubernetes/enhancements/issues/3715)
+* [Allow StatefulSet to control start replica ordinal numbering](	https://github.com/kubernetes/enhancements/issues/3335)
+* [Random Pod selection on ReplicaSet downscaling](	https://github.com/kubernetes/enhancements/issues/2185)
+    
+    
+### Deprecations and Removals 
```

Comment:
Please add a new removal chapter with something like this (feel free to rephrase):

#### InTreePluginXXXUnregister feature gates removed.

Alpha feature gates `InTreePluginAWSUnregister`, `InTreePluginAzureDiskUnregister`, `InTreePluginAzureFileUnregister`, `InTreePluginGCEUnregister`, `InTreePluginOpenStackUnregister` and `InTreePluginvSphereUnregister` were removed in Kubernetes 1.31.

These feature gates allowed us to test situation when the in-tree volume plugins were removed from code without actually removing them. The mentioned volume plugins were removed already in Kubernetes 1.30 and these feature gates did nothing useful.

The only remaining CSI migration gate is `InTreePluginPortworxUnregister` and it remains alpha until the CSI migration for Portworx is complete and the in-tree volume plugin is ready for removal.

\<end of suggested content>

* PR that removed the featgure gates: https://github.com/kubernetes/kubernetes/pull/124815 
* Issue about missing documentation: https://github.com/kubernetes/kubernetes/issues/126406

Reply:
+1 from me; this is important to explain.